src/chat/chatSelectors.js

```diff
@@ -13,3 +16,16 @@ export const getPointer = (state) => {
 
   return [messages[0].id, messages[messages.length - 1].id];
 };
+
+export const getRecentConversations = (state) => {
+  const selfEmail = state.account[0].email;
+  const messages = state.chat.messages[JSON.stringify(specialNarrow('private'))] || [];
+  const recipientSet = messages.reduce((accumulator, x) => {
```

Comment:
Reduce returns a single value, which can be an array. In fact `map` and `filter` can both be implemented via `reduce` and that is a valid use.

Still I went with a version similar to your 2nd suggestion as it is simpler.